src/libs/ship/src/ship_lights.cpp

```diff
@@ -655,13 +643,26 @@ void ShipLights::Realize(uint32_t dwDeltaTime)
 
 void ShipLights::UnSetLights(VAI_OBJBASE *pObject)
 {
+	if(!bLoadLights) return;
     for (int32_t i = iMinLight; i <= iMaxLight; i++)
         pRS->LightEnable(i, false);
 
     iMinLight = 1000;
     iMaxLight = -1;
 }
 
+void ShipLights::ResetLights(VAI_OBJBASE *pObject, bool bLight)
+{
+	for (uint32_t i = 0; i < aLights.size(); i++)
+	{
+		ShipLight &L = aLights[i];
+		if (L.bOff)
+            continue;
+		if(bLight) 	L.bTmpOff = false;
+		else		L.bTmpOff = true;
+	}		
```

Comment:
```suggestion
    for (auto &light : aLights)
    {
        if (!light.bOff)
        {
            light.bTmpOff = !bLight;
        }
    }		
```
would be more expressive